custom_components/healthchecksio/config_flow.py

```diff
@@ -78,10 +78,9 @@ async def _show_initial_config_form(self, user_input):
             step_id="user", data_schema=vol.Schema(data_schema), errors=self._errors
         )
 
-    async def async_step_self_hosted(
-        self, user_input={}
-    ):  # pylint: disable=dangerous-default-value
+    async def async_step_self_hosted(self, user_input=None):
```

Comment:
```suggestion
    async def async_step_self_hosted(self, user_input):
```

This can not be empty/none